simulation/test_http_connection.cpp

```diff
@@ -598,6 +598,55 @@ TORRENT_TEST(http_connection_http_error)
 	test_proxy_failure(settings_pack::http);
 }
 
+// Requests a proxied SSL connection. This test just ensures that the correct CONNECT request
+// is sent to the proxy server.
+TORRENT_TEST(http_connection_ssl_proxy)
+{
+	using sim::asio::ip::address_v4;
+	sim_config network_cfg;
+	sim::simulation sim{network_cfg};
+
+	sim::asio::io_service client_ios(sim, address_v4::from_string("10.0.0.1"));
+	sim::asio::io_service server_ios(sim, address_v4::from_string("10.0.0.2"));
+	sim::asio::io_service proxy_ios(sim, address_v4::from_string("50.50.50.50"));
+	lt::resolver res(client_ios);
+
+	sim::http_server http(server_ios, 8080);
+	sim::http_server http_proxy(proxy_ios, 4445);
+
+	lt::aux::proxy_settings ps = make_proxy_settings(settings_pack::http);
+
+	int handler_counter = 0;
+	int server_counter = 0;
+
+	http_proxy.register_handler("/10.0.0.2:8080"
```

Comment:
A slash is automatically added by the http server, in `normalize` (https://github.com/arvidn/libsimulator/blob/9e1948731b42b32b41e14ba3f16bd828d53c8a42/src/http_server.cpp#L97), so it won't match the request without the slash.